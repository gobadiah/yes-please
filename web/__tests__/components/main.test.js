import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Main from '../../components/main';

describe('Main component', () => {
  it('should allow for one child', () => {
    const main = shallow(<Main><span>Hello world</span></Main>);
    expect(main.find('span').text()).toEqual('Hello world');
  });

  it('should allow for multiple children', () => {
    const main = shallow(<Main><span>Hello</span><span>world</span></Main>);
    expect(main.find('span').length).toBe(2);
  });

  it('should match snapshot', () => {
    const component = renderer.create(<Main><span>Hello dude</span></Main>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
