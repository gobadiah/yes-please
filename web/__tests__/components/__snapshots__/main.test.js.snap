// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should match snapshot 1`] = `
<div>
  <span>
    Hello dude
  </span>
</div>
`;
